### PR TITLE
Js fill fe values

### LINKED ISSUE #887: FE System and FE simplification

### PR DESCRIPTION
**Pull request by jstebel**
Move fill_fe_values() to FEValues.
Remove template parameter spacedim from FiniteElement.

Partially resolves #887.

### CODE REVIEW

File: src/fem/finite_element.hh [hunk 1]
```diff
@@ -343,7 +352,7 @@ protected:
      * @param q Quadrature rule.
      * @param flags Update flags.
      */
-    virtual FEInternalData *initialize(const Quadrature<dim> &q);
+    FEInternalData *initialize(const Quadrature<dim> &q);
```
Comment (jbrezmorf) [inline src/fem/finite_element.hh]: Move this to FeValues together with FeInternalData.

File: src/fem/finite_element.hh [hunk 2]
```diff
@@ -299,7 +308,7 @@ public:
      * @param p    Point of evaluation.
      * @param comp Number of vector component.
      */
-    virtual double basis_value(const unsigned int i,
+    double basis_value(const unsigned int i,
```
Comment (jbrezmorf) [inline src/fem/finite_element.hh]: Rename and reimplement to shape_value. Same for basis_grad. Called just once per assembly loop so until we need to call it many times (in XFEM) we can live with slightly reduced efficiency.

File: src/fem/finite_element.hh (hunk 1 above)
Comment (jstebel) [reply src/fem/finite_element.hh]: Done.

File: src/fem/finite_element.hh (hunk 2 above)
Comment (jstebel) [reply src/fem/finite_element.hh]: Done.

**Pull request merged**